internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -412,16 +412,8 @@ func (s *Server) parseGithubEventForProcessing(
 		return fmt.Errorf("error getting repo information from payload: %w", err)
 	}
 
-	ph, err := s.store.GetParentProjects(ctx, dbRepo.ProjectID)
-	if err != nil {
-		return fmt.Errorf("error getting project hierarchy: %w", err)
-	}
-
 	// get the provider for the repository
-	prov, err := s.store.GetProviderByName(ctx, db.GetProviderByNameParams{
-		Name:     dbRepo.Provider,
-		Projects: ph,
-	})
+	prov, err := s.providerStore.GetByName(ctx, dbRepo.ProjectID, dbRepo.Provider)
```

Comment:
Genuine question: The old code was calling GetParentProjects and then getting the provider by name in parent. The new code goes to GetByName, which calls findProjects which calls getProjectHierarchy. Are those equivalent?

Reply:
`getProjectHierarchy` also calls `GetParentProjects`. In fact, I may get rid of the separate `getProjectHierarchy` function since it's only called in one places (originally in this branch it was called more than once)

Reply:
That being said - one important difference is that the `GetByName` will check to see if there are any duplicate results and return an error if there is ambiguity about being what is returned, whereas the code I have replaced will not (see the implementation of `GetByName`).

@eleftherias - since I think you wrote this code: is there any possibility that the original query could match more than one provider?

Reply:
It is possible, but only in a strange corner case where someone renamed their GitHub organization to have the same name as a GitHub organization that they previously installed Minder on. We generally don't handle renames yet, so I think we can ignore that case for now.